code/clue.js

```diff
@@ -1,59 +1,267 @@
+  'billiardroom', 
+  'lounge', 
+  'ballroom', 
+  'hall', 
+  'spa', 
+  'livingroom', 
+  'observatory', 
+  'theatre', 
+  'guesthouse', 
+  'patio'
+]
 
-const rooms = []
+// Selects randomly one item from the array passed into the function
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
-}
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
+// Creates an object that stores the mystery 
 // The values will be set later.
 
-// This function will be invoked when you click on the killer card.
+const mystery = {
+  killer: undefined,
+  weapon: undefined,
+  room: undefined,
+};
+
+// FUNCTION - this function will run when you click on the killer card.
+
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  document.getElementById('loader1').style.display = 'block';
+
+  setTimeout(afterLoaderPickKiller, 1600);
+};
+  
+document.getElementById('killerImage').style.filter = 'sepia(100%)';
+document.getElementById('loader1').style.display = 'none'; // prevents loader to begin when page is loaded
+document.getElementById('killerCard').addEventListener('click', pickKiller); // allows us to connect en event when clicked to the killer card
+
+// FUNCTION - this function runs when loader is done.
+
+const afterLoaderPickKiller = () => {
+  document.getElementById('killerImage').style.filter = 'sepia(0%)';
+  document.getElementById('loader1').style.display = 'none';
+  document.getElementById('killerCard').removeEventListener('click', pickKiller);
+
+// FUNCTION - this function randomly changes each killer's favourite weapon.
+
+  const shuffleFavouriteWeapon = () => {
+    mystery.favouriteWeapon = randomSelector(weapons)
+  }
+ 
+// This calls/executes the above function.
+
+  shuffleFavouriteWeapon();
+
+// These connects the js and the html, and outputs in browser.
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.favouriteWeapon.name}`
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}` 
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age}` 
```

Comment:
`document.getElementById('killerAge').innerHTML = mystery.killer.age`  also works here since you aren't concatenating strings together